go/test/endtoend/vtgate/trxn_mode_test.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vtgate
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+)
+
+var (
+	clusterInstance *cluster.LocalProcessCluster
+	vtParams        mysql.ConnParams
+	keyspaceName    = "ks"
+	cell            = "zone1"
+	hostname        = "localhost"
+	sqlSchema       = `
+	create table twopc_user (
+		user_id bigint,
+		name varchar(128),
+		primary key (user_id)
+	) Engine=InnoDB;
+
+	create table twopc_lookup (
+		name varchar(128),
+		id bigint,
+		primary key (id)
+	) Engine=InnoDB;`
+
+	vSchema = `
+	{	
+		"sharded":true,
+		"vindexes": {
+			"hash_index": {
+				"type": "hash"
+			},
+			"twopc_lookup_vdx": {
+				"type": "lookup_hash_unique",
+				"params": {
+				  "table": "twopc_lookup",
+				  "from": "name",
+				  "to": "id",
+				  "autocommit": "true"
+				},
+				"owner": "twopc_user"
+			}
+		},	
+		"tables": {
+			"twopc_user":{
+				"column_vindexes": [
+					{
+						"column": "user_id",
+						"name": "hash_index"
+					},
+					{
+						"column": "name",
+						"name": "twopc_lookup_vdx"
+					}
+				]
+			},
+			"twopc_lookup": {
+				"column_vindexes": [
+					{
+						"column": "id",
+						"name": "hash_index"
+					}
+				]
+			}
+		}
+	}
+	`
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	exitCode := func() int {
+		clusterInstance = &cluster.LocalProcessCluster{Cell: cell, Hostname: hostname}
+		//Set extra tablet args for twopc
+		clusterInstance.VtTabletExtraArgs = []string{
+			"-twopc_enable",
+			"-twopc_coordinator_address", "localhost:15028",
```

Comment:
Get the port number from clusterInstance.VtgateGrpcPort

Reply:
Done